$NetBSD: patch-aa,v 1.21 2003/08/13 13:24:16 atatat Exp $

If LSOF_MORE_SECURE is non-zero, compile lsof with security, and
only let unprivileged (non-root) users view their own open files.
Set LSOF_LESS_SECURE_SOCKETS to non-zero in order to allow non-root
users to see open sockets, even when LSOF_MORE_SECURE is set.

Patch for vnode/DNLC interface change courtesy of lsof maintainer.

--- dialects/n+obsd/machine.h.orig	Tue Dec  3 13:23:54 2002
+++ dialects/n+obsd/machine.h
@@ -214,16 +214,12 @@
  * that lsof can search.  A value of 1 directs printname() to prefix the
  * cache value with the file system directory name; 2, avoid the prefix.
  *
- * HASNCAPID is defined for those dialects with a searchable kernel name
- * cache whose cache and vnodes are linked by a capability ID.
- *
  * NCACHELDPFX is a set of C commands to execute before calling ncache_load().
  *
  * NCACHELDSFX is a set of C commands to execute after calling ncache_load().
  */
 
 #define	HASNCACHE	1
-#define	HASNCAPID	1
 /* #define	NCACHELDPFX	??? */
 /* #define	NCACHELDSFX	??? */
 
@@ -371,7 +371,9 @@
  * (the one that its user logged on with) of the lsof process.
  */
 
-/* #define	HASSECURITY	1 */
+#if LSOF_MORE_SECURE
+#define	HASSECURITY	1
+#endif
 
 
 /*
@@ -380,7 +382,9 @@
  * listing is selected by the "-i" option.
  */
 
-/* #define	HASNOSOCKSECURITY	1	*/
+#if LSOF_LESS_SECURE_SOCKETS
+#define	HASNOSOCKSECURITY	1
+#endif
 
 
 /*
@@ -497,7 +493,7 @@
 
 # if	(defined(OPENBSDV) && OPENBSDV>=2010) || (defined(NETBSDV) && NETBSDV>=1020)
 #define	USE_LIB_RNMH				1	/* rnmh.c */
-# else	/* (defined(OPENBSDV) && OPENBSDV>=2010) || (defined(NETBSDV) && NETBSDV>=1020) */
+# else	/* (defined(OPENBSDV) && OPENBSDV<2010) && (defined(NETBSDV) && NETBSDV<1020) */
 #define	USE_LIB_RNAM				1	/* rnam.c */
 # endif	/* (defined(OPENBSDV) && OPENBSDV>=2010) || (defined(NETBSDV) && NETBSDV>=1020) */
 
